src/components/Select/index.tsx

```diff
@@ -115,27 +125,32 @@ export const Select: React.FC<SelectProps> = ({
   ...props
 }) => {
   const { isMobile } = useDeviceInfo()
-  const [showCustomSelect, setShowCustomSelect] = React.useState(false)
-  const [isOpen, setIsOpen] = React.useState(false)
-  const [selectedItem, setSelectedItem] = React.useState(
+  const [showCustomSelect, setShowCustomSelect] = useState(false)
+  const [isOpen, setIsOpen] = useState(false)
+  const [query, setQuery] = useState('')
+  const [selectedItem, setSelectedItem] = useState(
     initialSelectedItem || items[0]
   )
-  const [highlightedIndex, setHighlightedIndex] = React.useState(
+  const [highlightedIndex, setHighlightedIndex] = useState(
     items.indexOf(initialSelectedItem || items[0])
   )
-  const containerRef = React.useRef<HTMLDivElement>(null)
-  const inputRef = React.useRef<HTMLInputElement>(null)
-  const menuRef = React.useRef<HTMLUListElement>(null)
+  const containerRef = useRef<HTMLDivElement>(null)
+  const inputRef = useRef<HTMLInputElement>(null)
+  const menuRef = useRef<HTMLUListElement>(null)
   let itemRefs: HTMLLIElement[] = []
   const labelId = `${id}-label`
   const menuId = `${id}-menu`
   const esc = useKeyPress('Escape')
   const arrowUp = useKeyPress('ArrowUp')
   const arrowDown = useKeyPress('ArrowDown')
+  const resetQuery = useCallback(
+    debounce(() => setQuery(''), 1500),
```

Comment:
I think this is a bit too long. You are not waiting for 1.5 seconds until you type your next letter. Native selects in browsers have a bit shorter delay I think

Reply:
1000 😬

Reply:
I'll switch this to 1 second as defined in the HTML source code: [typeAheadTimeout](https://chromium.googlesource.com/chromium/blink.git/+/refs/heads/main/Source/core/html/forms/TypeAhead.cpp#45)